From 6613b87b4f4c229241f34b09aa42ae53a335e368 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:01:32 -0400
Subject: [ethernet] mlx4: Do not disable vlan filter during promiscuous mode

Message-id: <490f152f651b0fa2e99ff6a600cc31d8c02fcac2.1410425016.git.avadai@redhat.com>
Patchwork-id: 93690
O-Subject: [Patch RHEL7.1 V1 156/197] net/mlx4_en: Do not disable vlan filter during promiscuous mode
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Eugenia Emantayev <eugenia@mellanox.com>

Promiscous mode is only for MACs.
Should not disable/enable VLAN filter when entering/leaving promisuous mode.

Signed-off-by: Aviad Yehezkel <aviadye@mellanox.co.il>
Signed-off-by: Eugenia Emantayev <eugenia@mellanox.com>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 49a1e4f6b7e78fed9ab36ebbb9d88656885255a4)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
index 2bd78f6..518eb1b 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
@@ -940,11 +940,6 @@ static void mlx4_en_set_promisc_mode(struct mlx4_en_priv *priv,
        0, MLX4_MCAST_DISABLE);
   if (err)
    en_err(priv, "Failed disabling multicast filter\n");
-
-  /* Disable port VLAN filter */
-  err = mlx4_SET_VLAN_FLTR(mdev->dev, priv);
-  if (err)
-   en_err(priv, "Failed disabling VLAN filter\n");
  }
 }
 
@@ -993,11 +988,6 @@ static void mlx4_en_clear_promisc_mode(struct mlx4_en_priv *priv,
    en_err(priv, "Failed disabling promiscuous mode\n");
   break;
  }
-
- /* Enable port VLAN filter */
- err = mlx4_SET_VLAN_FLTR(mdev->dev, priv);
- if (err)
-  en_err(priv, "Failed enabling VLAN filter\n");
 }
 
 static void mlx4_en_do_multicast(struct mlx4_en_priv *priv,
-- 
1.7.1